spark/src/main/java/org/apache/zeppelin/spark/ZeppelinContext.java

```diff
@@ -220,7 +222,8 @@ public static String showDF(SparkContext sc,
       Object df, int maxResult) {
     Object[] rows = null;
     Method take;
-    String jobGroup = "zeppelin-" + interpreterContext.getParagraphId();
+    String jobGroup = "zeppelin-" + interpreterContext.getNoteId() + "-"
+        + interpreterContext.getParagraphId();
```

Comment:
I see several places of constructing jobGroupId from noteId and extract noteId from jobGroupId. It's better to put them together so that others can understand it easily and other places can reuse it.